proposed/template-renderer.md

```diff
@@ -0,0 +1,209 @@
+Common Interface for Rendering Templates
+========================================
+
+This document describes a common interface for template renderers.
+
+The goal set by `TemplateRendererInterface` is to standardize how frameworks, libraries and CMSs
+render their template so that projects are more free to use the template renderer they prefer.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119][].
+
+The word `implementor` in this document is to be interpreted as someone
+implementing the `TemplateRendererInterface` in a [template engine][] library.
+Users of template renderer are referred to as `user`.
+The word `enduser` in this document is to be interpreted as someone using
+a library, framework, cms created by a `user`.
+
+[RFC 2119]: http://tools.ietf.org/html/rfc2119
+
+## Goal
+
+Having common interfaces for rendering templates allows developers to create libraries that can interact with many frameworks and other libraries in a common fashion.
+
+Some examples may use the Interface:
+
+ - Content Management Systems (Sulu CMS, Drupal, Typo3, Contao CMS, ...)
+ - E-Commerce Platforms (Sylius, Spryker)
+ - Newsletter/Mail Libraries (Symfony Mailer, Abstractions over Mailchimp and other Newsletter tools) 
+ - Anything following ["Separating content from presentation"](https://en.wikipedia.org/wiki/Separation_of_content_and_presentation) where the presentation is project specific.
+
+Some examples may implement the Interface or providing a bridge:
```

Comment:
Yii has alike interface https://github.com/yiisoft/view/blob/master/src/TemplateRendererInterface.php with the difference that we're passing a rendering context as `$this` for nested template rendering.

Reply:
Looking at the example it works the same way `->render($template, $params)`: https://github.com/yiisoft/view/blob/master/docs/basic-functionality.md#rendering

That yii uses then in the `template` `$this` is totally fine the TemplateRendererInterface doesn't force how the the variables need to access in the template. If it is `{variable}`, `{{variable}}` or `<?php echo $this->variable ?>` is up to the template renderer.

Thx for mentioning yii will add it also the the list. Please check my details in the PR description about it if I did write nothing wrong about it.